front/src/js/service.js

```diff
@@ -56,19 +56,47 @@ function ScratchyService(apiUrl) {
 
 //                               ROOM FUNCTIONS
 
+    /**
+     *
+     * @async
+     * @augments ScratchyService
+     * @param {Room} room - room in which
+     * @param {User} user - user to add in room 
+     * @returns {Promise<undefined>} - 
+     */
+    this.addUserToRoom = async function(room,user){
+        if ( !room.users.includes(user.id)) {
+            room.users.push(user.id);
+            this.updateRoom(room);
+        }else{
+            console.log("user is already in the room");
+        }
+    };
+    
+    /**
+     *
+     * @async
+     * @augments ScratchyService
+     * @param {Room} room - room in which
+     * @param {User} user - user to add in room 
+     * @returns {Promise<undefined>} - 
+     */
+    this.removeUserInRoom = async function(room,user){
+        room.users = room.usersID.filter( uid => uid != user.id );
+        this.updateRoom(room);
```

Comment:
almost there,
`room.users = room.usersID...` should be `room.users = room.users...`
`user to add in room` should be `user to remove from room`